horace_core/sqw/file_io/@faccess_sqw_v4/get_exp_info.m

```diff
@@ -35,7 +35,8 @@
     if ~isinf(samp_inst_number)
         exp_data = exp_data(samp_inst_number);
     end    
-    exper = Experiment(IX_detector_array(),IX_null_inst(),IX_null_sample,exp_data);
+    % leave detector arrays to be filled in from detpar
+    exper = Experiment([],IX_null_inst(),IX_null_sample,exp_data);
     return;
```

Comment:
This is not good. If I am getting `experiment_info` from file, I would expect it to contain det_info, if it is there from now. 
Why to connect them through `sqw`? 

I expect that future work is planned to fix this issue, and the method will return full experiment info with detector arrays and sqw.detpar will be referring there.

Reply:
I'm going to agree with this - I have made this change elsewhere - and look at why it is not changed here

Reply:
Having said I agree with this point in general, the way that get_exp_info is used in the code is either (a) a stand-alone call in various tests, where once the Experiment is returned, its contents other than detector arrays are tested, (b) extract Experiment data to check it in the course of some algorithm, (c) general read process, where get_exp_info is followed or preceded by get_detpar. In these cases, either the detector info is not required or it is being got separately in the immediate vicinity of the call.

the purpose of this change is to prevent a spurious default IX_detector_array being introduced into the Experiment.  - it is easier to check an empty IX_detector_array than a default one. If the detector arrays had been saved into the Experiment in the v4 format rather than being routed into detpar, something might have been done about it. As it is, this method works in all available cases and a refactor to create a new file format and avoid using detpar should fix the problem later.